FhirToDataLake/SynapseValidate/Synapse_Validator.py

```diff
@@ -0,0 +1,207 @@
+import os
+import sys
+import argparse
+import concurrent.futures
+import json
+import requests
+import pyodbc
+import pandas as pd
+import time
+from urllib import parse as urlparse
+from datetime import datetime
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--SynapseWorkspaceName", help="Name of Synapse workspace.")
+parser.add_argument("--FhirServerUrl", help="Fhir server url.")
+parser.add_argument("--Database", help="Name of database.")
+parser.add_argument("--SchemaCollectionDirectory", help="Schema collection directory path.")
+args = parser.parse_args()
+
+Sql_Server_Endpoint = args.SynapseWorkspaceName + "-ondemand.sql.azuresynapse.net"
+Database = args.Database
+Schema_Collection_Directory = args.SchemaCollectionDirectory
+Fhir_Server_Base_Url = args.FhirServerUrl
+Sql_Username = os.getenv('SQL_USERNAME')
+Sql_Password = os.getenv('SQL_PASSWORD')
+
+pyodbc.pooling = False
+
+class RawDataSource:
+    Headers = {'Accept': 'application/fhir+json', "Prefer": "respond-async"}
+    Start_Datetime = "1970-01-01T00:00:00-00:00"
+
+    def __init__(self, baseUrl):
+        self.baseUrl = baseUrl
+
+    def get_all_entries(self, resource_type):
+        end_datetime = datetime.today().strftime('%Y-%m-%dT%H:%M:%S-00:00')
+        url = self.baseUrl + f"/{resource_type}?_lastUpdated=ge{RawDataSource.Start_Datetime}&_lastUpdated=lt{end_datetime}&_sort=_lastUpdated&_count=1000"
+        historical_data = self.get_entries(url)
+
+        '''
+        url = self.baseUrl + f"/{resource_type}?_lastUpdated=ge{end_datetime}&_lastUpdated=lt{end_datetime}&_sort=_lastUpdated&_count=1000"
+        latest_data = self.get_entries(url)
+        return historical_data + latest_data
+        '''
+        return historical_data
+
+
+    def get_entries(self, url):
```

Comment:
It looks to me that we don't need to extract this as an independent function. We can put all logics to get resources into a single function like `get_all_resources`.

Reply:
Sure, only use **get_all_entries()** in FhirApiDataClient, because response body use "entry" to represent resources.